_episodes/15-coding-conventions.md

```diff
@@ -431,11 +433,11 @@ because an incorrect comment causes more confusion than no comment at all.
 > ~~~
 >  {: .language-bash}
 >
-> Next look at the `catchment-analysis.py` file in PyCharm
+> Next look at the `catchment-analysis.py` file in VS Code
 > and identify where the above guidelines have not been followed.
 > Fix the discovered inconsistencies and commit them to the feature branch.
 >> ## Solution
->> Modify `catchment-analysis.py` from PyCharm, 
+>> Modify `catchment-analysis.py` from VS Code, 
 >> which is helpfully marking inconsistencies with coding guidelines by underlying them.
```

Comment:
VS Code doesn't mark inconsistencies in the code like PyCharm does (see https://www.reddit.com/r/vscode/comments/oyhfq9/is_there_an_extension_that_encourages_pep8/). We are going to have to look for a solution here for the students to install, or re-word / remove this exercise.

Reply:
Agreed, we need to get this sorted.

Reply:
I use pylint in VS-Code, it does a pretty good job of marking the issues in this exercise. We'll need to add a point where we get the students to:
- Go to extensions
- Search for "Pylint"
- Install Pylint
Then they'll need to make sure it is active, which it should be by default. Might need to close VS-Code and reopen to get it working on the file when making modifications. By default, maximum line length is 100 characters. Otherwise I *think* PyLint highlights what's needed for this exercise out the box, although it does seem to miss some pep8 style guidelines.

Reply:
I've tested Flake8 as well - it seems to do a better job of sticking to pep8: it picks out when there are incorrect spaces around "=" or inside arguments, and its default max line length is 80. Suggest we recommend students install Flake8, noting Pylint is another commonly used option.

Reply:
See new issue: https://github.com/carpentries-incubator/python-intermediate-development-earth-sciences/issues/40

Reply:
I have added instructions to add the Flake8 extension, and added to this branch @Scottan